match-api/src/index.ts

```diff
@@ -12,153 +12,227 @@ const expressJwt = require("express-jwt");
 const prisma = new PrismaClient();
 const app = express();
 
-app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.urlencoded({extended: true}));
 app.use(bodyParser.json());
 app.use(express.json());
 app.use(cookieParser());
 app.use(
-  expressJwt({
-    secret: process.env.SECRET,
-    algorithms: ["HS256"],
-    getToken: (req: any) => req.cookies.token,
-  }).unless({ path: ["/ping", "/signup", "/login"] })
+    expressJwt({
+        secret: process.env.SECRET,
+        algorithms: ["HS256"],
+        getToken: (req: any) => req.cookies.token,
+    }).unless({path: ["/ping", "/signup", "/login"]})
 );
 
 app.use(
-  (
-    err: { name: string; status: any; message: any },
-    req: any,
-    res: any,
-    next: () => void
-  ) => {
-    if (err.name === "UnauthorizedError") {
-      res.status(err.status).send({ message: err.message });
-      console.log("Auth middleware", "Auth error");
-      return;
+    (
+        err: { name: string; status: any; message: any },
+        req: any,
+        res: any,
+        next: () => void
+    ) => {
+        if (err.name === "UnauthorizedError") {
+            res.status(err.status).send({message: err.message});
+            console.log("Auth middleware", "Auth error");
+            return;
+        }
+        next();
     }
-    next();
-  }
 );
 
+app.put("/match/join/:id", async (req, res) => {
+    const {id} = req.params;
+    const user = (req as any).user;
+    try {
+        const matchData = await prisma.match.findUnique({
+            where: {id: Number(id)},
+            include: {
+                Invitation: true
+            }
+        });
+        if (!matchData) {
+            return res.sendStatus(404);
```

Comment:
why not throwing an error to get similar to line 59 ?

Reply:
je pense que c'est mieux de suivre la norme http